FRFuel.cs

```diff
@@ -88,350 +58,156 @@ public FRFuel()
             {
-                return;
-            }
-
-            Vector3 pos = GetEntityCoords(PlayerPedId(), true);
-
-            int model = 883325847;
-
-            Function.Call(Hash.REQUEST_MODEL, model);
-
-            IEnumerable<Vector3> positions = GetNearbyJerryCanPickUpCoordinates(pos);
+                var replaceCans = _jerryCans.Where(p => p.IsCollected).ToList();
+                _jerryCans.RemoveAll(x => replaceCans.Any(a => a.Handle == x.Handle));
 
-            if (positions.Count() == 0 && pickups.Count != 0)
-            {
-                pickups.ForEach(p => p.Delete());
-                pickups.Clear();
-            }
-            else
-            {
-                positions.ToList().ForEach(position =>
+                foreach (var pickup in replaceCans)
                 {
-                    if (!pickups.Any(pickup => position.DistanceToSquared(pickup.Position) < 5f))
-                    {
-                        // add pickup if one doesn't exist within 5f proximity of it
-                        int pickupHandle = CreatePickup(
-                            0xc69de3ff, // Petrol Can
-                            position.X, position.Y, position.Z - 0.5f,
-                            8 | 32, // Place on the ground, local only
-                            0,
-                            true,
-                            (uint)model);
-
-                        Pickup pickup = new Pickup(pickupHandle);
-
-                        pickups.Add(pickup);
-                    }
-                });
-            }
-            await Task.FromResult(0);
-        }
-        #endregion
-
-        /// <summary>
-        /// External API for getting current fuel leve;
-        /// </summary>
-        /// <returns></returns>
-        public float ExportsGetCurrentFuelLevel()
-        {
-            if (!PlayerVehicleViableForFuel())
-            {
-                return -1f;
-            }
-
-            return Game.PlayerPed.CurrentVehicle.FuelLevel;
-        }
-
-        /// <summary>
-        /// External API for adding fuel
-        /// </summary>
-        /// <param name="amount"></param>
-        public void ExportsAddFuel(float amount)
-        {
-            if (PlayerVehicleViableForFuel())
-            {
-                var vehicle = Game.PlayerPed.CurrentVehicle;
-
-                VehicleSetFuelLevel(vehicle, vehicle.FuelLevel + amount);
-            }
-        }
-
-        /// <summary>
-        /// External API for setting fuel level
-        /// </summary>
-        /// <param name="amount"></param>
-        public void ExportsSetFuel(float amount)
-        {
-            if (PlayerVehicleViableForFuel())
-            {
-                var vehicle = Game.PlayerPed.CurrentVehicle;
+                    var pos = pickup.Position;
+                    pickup.Delete();
+                    _jerryCans.Add(new Pickup(API.CreatePickup(0xc69de3ff, pos.X, pos.Y, pos.Z - 0.5f, 8 | 32, 0, true, 883325847)));
```

Comment:
Honestly i don't like that you introduced one more "magic" number (instead of keeping named variable) and stripped comment.